lib/components.js

```diff
@@ -734,14 +734,16 @@ export class List extends Component {
 
   // If there is an index go to the detail view of that item
   // otherwise create a new record
-  genNew(index){
+  // if a value is passed in, that will override an existing value
+  genNew(index, value=null){
     const { options, ctx } = this.props;
     const auto = this.getAuto();
     const i18n = this.getI18n();
     const config = this.getConfig();
     const stylesheet = this.getStylesheet();
     const templates = this.getTemplates();
-    const itemValue = Number.isInteger(index) ? this.state.value[index] : null;
+    const initialValue = value ? value : this.state.value[index]
```

Comment:
I think this could be expressed more cleanly with a traditional if/ else.

```
if (value) {
    const itemValue = value
} else {
    const itemValue = Number.isInteger(index) ? this.state.value[index] : null
}
```

Might make more sense to name the sent-in parameter to something like `passedInValue` or `sentValue` or something that indicates it came from the caller.

Reply:
Cool - I liked passedInValue - I adjusted the if/else logic a bit from your example, but same general idea.